common/src/terraingen.rs

```diff
@@ -0,0 +1,996 @@
+use crate::world::Material;
+
+pub struct VoronoiInfo {
+    pub location: na::Vector2<f64>,
+    pub material: Material,
+}
+impl VoronoiInfo {
+    pub fn new(mat: Material, rain: f64, temp: f64) -> VoronoiInfo {
+        VoronoiInfo {
+            location: na::Vector2::new(rain, temp),
+            material: mat,
+        }
+    }
+
+    pub fn terraingen_voronoi(elev: f64, rain: f64, temp: f64) -> Material {
+        let voronoi_choices_high = vec![
```

Comment:
Every time you call `Vec`, heap memory is allocated. Given that this is constant data, allocating a fresh copy for every single voxel(!) doesn't make sense. Instead, these could be `const` arrays (in which case `new` will have to be a `const fn`). For obscure annoying reasons `na::Vector2` can't be used in constants, so you'll have to replace it with a `[f32; 2]` or similar, but that shouldn't be too much trouble.

Reply:
Made this change. I've also combined the terraingen_voronoi and surface_adjuster functions into one function which now takes an extra parameter dist representing depth below the ground surface (not the guiding plane).